src/addrdb.h

```diff
@@ -44,31 +36,17 @@ class CBanEntry
         nCreateTime = nCreateTimeIn;
     }
 
-    explicit CBanEntry(int64_t n_create_time_in, BanReason ban_reason_in) : CBanEntry(n_create_time_in)
+    SERIALIZE_METHODS(CBanEntry, obj)
     {
-        banReason = ban_reason_in;
+        uint8_t ban_reason = 2; //! For backward compatibility
+        READWRITE(obj.nVersion, obj.nCreateTime, obj.nBanUntil, ban_reason);
```

Comment:
Note for reviewers: this means that if any peers are discouraged at the time that the node was last shut down pre-upgrade, they'll be loaded as banned post-upgrade. They'll clear automatically after 24 hours, so I think this is ok.